src/udev/net/ethtool-util.c

```diff
@@ -206,3 +212,108 @@ int ethtool_set_wol(int *fd, const char *ifname, WakeOnLan wol) {
 
         return 0;
 }
+
+static int ethtool_get_stringset(int *fd, struct ifreq *ifr, int stringset_id, struct ethtool_gstrings **gstrings) {
+        _cleanup_free_ struct ethtool_gstrings *strings = NULL;
+        struct ethtool_sset_info info =  {
+                .cmd = ETHTOOL_GSSET_INFO,
+                .reserved = 0,
+                .sset_mask = 1ULL << stringset_id,
+        };
+        unsigned len;
+        int r;
+
+        ifr->ifr_data = (void *) &info;
```

Comment:
all ptr types implicitly cast into `void*`, hence this explicitly cast is unnecessary.